### PR TITLE
mso_user modified

### LINKED ISSUE #45: [mso_user] Add support for specifying read-only roles
Add read-only and read-write per role capability for each role while maintaining backward compatibility.

### PR DESCRIPTION
**Pull request by shrsr**
1. Updated module to capture and fail creation of a user with no password.
2. Updated test file to include read only case.

Resolves #45

### CODE REVIEW

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 1]
```diff
@@ -116,6 +130,121 @@
     - nm_add_user_again.current.id == nm_add_user.current.id
     - nm_add_user_again.current.username == 'ansible_test'
 
+- name: Add user in read only mode mode (check_mode)
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:133]: "in read only mode mode" -> "with read only role"

File: plugins/modules/mso_user.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def main():
         payload = dict(
             id=user_id,
             username=user_name,
-            password=user_password,
+            # password=user_password,
```
Comment (lhercot) [inline plugins/modules/mso_user.py:231]: If it pass, remove the line.

File: plugins/modules/mso_user.py [hunk 3]
```diff
@@ -256,7 +259,10 @@ def main():
                     mso.existing = mso.proposed
                 else:
                     mso.existing = mso.request(path, method='PUT', data=mso.sent)
+
         else:
+            if user_password is None:
+                mso.fail_json("No password found. Please enter a password to create the user.")
```
Comment (lhercot) [inline plugins/modules/mso_user.py:265]: "The user {0} does not exist. The 'user_password' attribute is required to create a new user.".format(user_name)

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 4]
```diff
@@ -28,6 +28,18 @@
     user: ansible_test2
   register: cm_remove_user
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:29]: remove the register on all 3 in the clean environment part. You could combine all 3 with a loop.

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 5]
```diff
@@ -116,6 +130,121 @@
...
+    - cm_add_user2.current.id is not defined
+    - cm_add_user2.current.username == 'ansible_test_read'
+    - cm_add_user2.current.lastName == 'Test2'
+    - cm_add_user2.current.firstName == 'Ansible2'
+    - cm_add_user2.current.emailAddress == 'mso3@cisco.com'
+    - cm_add_user2.current.phoneNumber == '+32 478 436 299'
+    - cm_add_user2.current.accountStatus == 'active'
+    - cm_add_user2.current.roles[0].accessType == 'readOnly'
+
+- name: Add user in read only (normal mode)
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:172]: "in read only" -> "with read only role"

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 6]
```diff
@@ -116,6 +130,121 @@
...
+    - nm_add_user2.current.id is defined
+    - nm_add_user2.current.username == 'ansible_test_read'
+    - nm_add_user2.current.lastName == 'Test2'
+    - nm_add_user2.current.firstName == 'Ansible2'
+    - nm_add_user2.current.emailAddress == 'mso3@cisco.com'
+    - nm_add_user2.current.phoneNumber == '+32 478 436 299'
+    - nm_add_user2.current.accountStatus == 'active'
+    - nm_add_user2.current.roles[0].accessType == 'readOnly'
+
+- name: Add user in read only again(check mode)
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:189]: "in read only again" -> "with read only role again"

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 7]
```diff
@@ -116,6 +130,121 @@
...
+    <<: *user_present2
+    user_password: '#123455#123455Aa'
+  check_mode: yes  
+  register: cm_add_user2_again
+
+- name: Add user in read only again (normal mode)
+  mso_user: *user_present2
+  register: nm_add_user2
+
+- name: Add user3 in read only mode mode with no password (check_mode)
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:200]: "in read only mode mode" -> "with read only role"

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 8]
```diff
@@ -116,6 +130,121 @@
...
+    - nm_add_user2.current.roles[0].accessType == 'readOnly'
+
+- name: Add user in read only again(check mode)
+  mso_user: 
+    <<: *user_present2
+    user_password: '#123455#123455Aa'
+  check_mode: yes  
+  register: cm_add_user2_again
+
+- name: Add user in read only again (normal mode)
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:196]: "in read only again" -> "with read only role again"

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 9]
```diff
@@ -116,6 +130,121 @@
...
+    state: present
+  ignore_errors: yes
+  register: nm_add_user3
+
+- name: Verify nm_add_user2
+  assert:
+    that:
+    - nm_add_user3.msg == "No password found. Please enter a password to create the user."
+
+- name: Add user3 in read only with password (normal mode)
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:230]: "in read only" -> "with read only role"

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 10]
```diff
@@ -274,6 +403,17 @@
     - nm_query_user.current.username == 'ansible_test'
     - cm_query_user == nm_query_user
 
+- name: Query our user
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:406]: Query our read-only user

File: tests/integration/targets/mso_user/tasks/main.yml (hunk 8 above)
Comment (lhercot) [reply tests/integration/targets/mso_user/tasks/main.yml:182]: Seems to still be here

File: tests/integration/targets/mso_user/tasks/main.yml [hunk 11]
```diff
@@ -274,10 +389,30 @@
...
+- name: Verify query_user2
+  assert:
+    that:
+    - nm_query_user2 is not changed
+    - nm_query_user2.current.roles[0].accessType == 'readOnly'
 
 # REMOVE USER
 - name: Remove user (check_mode)
-  mso_user: *user_absent
+  mso_user:
```
Comment (lhercot) [inline tests/integration/targets/mso_user/tasks/main.yml:406]: You should add an anchor here and use it for the next 3 steps.
Like :
mso_user:  &user_absent

File: tests/integration/targets/mso_user/tasks/main.yml (hunk 8 above)
Comment (shrsr) [reply tests/integration/targets/mso_user/tasks/main.yml:182]: Weird. That line doesn't even exist on my branch

Comment (lhercot) [reply tests/integration/targets/mso_user/tasks/main.yml:182]: It is still here so we need to see how we can change that.

**Pull request merged**